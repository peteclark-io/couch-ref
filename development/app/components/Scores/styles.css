@import "../../css/branding.css";

.teams {
  font-family: var(--font-secondary-bold);
  font-size: 1.6rem;
  text-transform: uppercase;
  color: black;
}

.goals {
  font-family: var(--font-secondary-bold);
  font-size: 1.4rem;
  color: black;
  float: right;
}

.link {
  color: inherit;
  cursor: pointer;
}

.link:hover {
  text-decoration: none;
}

.match-list {
  list-style: none;
  padding-left: 0px;
}

.match-list li {
  margin-bottom: 18px;
}

.kick-off {
  line-height: normal;
  font-family: var(--font-secondary);
  font-size: 1rem;
}

.full-time {
  background: repeating-linear-gradient(
    45deg,
    white,
    white 10px,
    whitesmoke 10px,
    whitesmoke 20px
  );
}

.loading {
  font-size: 14rem;
  text-align: center;
}
