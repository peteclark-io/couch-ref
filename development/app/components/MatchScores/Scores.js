'use strict'

import React from 'react';
import _ from 'lodash';
import moment from 'moment';

import {Link} from 'react-router';
import {ThreeBounce} from 'better-react-spinkit';

import styles from './styles.css';
import Score from './Score';

const Scores = React.createClass({

   propTypes: {
      matches: React.PropTypes.arrayOf(
         React.PropTypes.shape({
            'id': React.PropTypes.string,
            'home': React.PropTypes.string,
            'away': React.PropTypes.string,
            'kickOff': React.PropTypes.object,
            'goalsHome': React.PropTypes.number,
            'goalsAway': React.PropTypes.number
         }))
      },

      render: function() {
         if (this.props.matches.length === 0){
            return (
               <div className={styles.loading}>
                  <ThreeBounce />
               </div>
            );
         }

         var today = moment(24, 'HH'); // midnight tonight
         var yesterday = moment(24, 'HH').subtract(1, 'days'); // midnight last night

         var todaysFixtures = _.filter(this.props.matches, (i) => {
            return i.kickOff.isBefore(today) && i.kickOff.isAfter(yesterday) && i.televised;
         });

         var title = 'Upcoming Fixtures';
         var fixtures = [];
         if (todaysFixtures.length > 0){
            fixtures = todaysFixtures;
            title = 'Today\'s Fixtures';
         } else {
            fixtures = _.filter(this.props.matches, (i) => {
               return i.kickOff.isAfter(today) && i.televised;
            });
         }

         fixtures = _.sortBy(fixtures, ['fullTime', 'kickOff']);

         if(fixtures.length === 0){
            return (
               <div>
                  <h2 className={styles['fixture-list-header']}>{title}</h2>
                  <div className={styles.spacer}></div>
                  <h4 className={styles['no-fixtures']}>No Fixtures right now! Check back later for new fixtures.</h4>
               </div>
            )
         }

         return (
            <div>
               <h2 className={styles['fixture-list-header']}>{title}</h2>
               <div className={styles.spacer}></div>

               <ul className={styles['match-list']}>
                  {fixtures.map(function(match) {
                     return (
                        <li key={match.id}>
                           <Score match={match} />
                        </li>
                     );
                  })}
               </ul>
            </div>
         );
      }
   });

   export default Scores;
