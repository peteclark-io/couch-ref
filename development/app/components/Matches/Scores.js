'use strict'

import React from 'react';
import styles from './styles.css';
import Score from './Score';
import {Link} from 'react-router';

const Scores = React.createClass({

  propTypes: {
    matches: React.PropTypes.arrayOf(
      React.PropTypes.shape({
        'id': React.PropTypes.string,
        'home': React.PropTypes.string,
        'away': React.PropTypes.string,
        'goalsHome': React.PropTypes.number,
        'goalsAway': React.PropTypes.number
      }))
  },

  render: function() {
    return (
      <ul className={styles['match-list']}>
        {this.props.matches.map(function(match) {
          return <li className={match.fullTime ? styles['full-time'] : ''} key={match.id}>
                    <Link className={styles.link} to={`/match/${match.id}`}>
                      <Score match={match} />
                    </Link>
                 </li>;
        })}
      </ul>
    );
  }
});

export default Scores;
