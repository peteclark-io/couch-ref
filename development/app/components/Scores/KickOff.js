'use strict'

import React from 'react';
import styles from './styles.css';

const KickOff = React.createClass({

  propTypes: {
    'kickOff': React.PropTypes.object
  },

  render: function() {
    return (
        <span className={styles['kick-off']} ref={node => (this.root = node)}>{this.props.kickOff.format("ddd Do MMM HH:mm")}</span>
    );
  }
});

export default KickOff;
