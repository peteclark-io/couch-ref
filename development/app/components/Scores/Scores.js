'use strict'

import React from 'react';
import _ from 'lodash';
import moment from 'moment';

import {Link} from 'react-router';
import {ThreeBounce} from 'better-react-spinkit';

import styles from './styles.css';
import Score from './Score';

const Scores = React.createClass({

  propTypes: {
    matches: React.PropTypes.arrayOf(
      React.PropTypes.shape({
        'id': React.PropTypes.string,
        'home': React.PropTypes.string,
        'away': React.PropTypes.string,
        'kickOff': React.PropTypes.object,
        'goalsHome': React.PropTypes.number,
        'goalsAway': React.PropTypes.number
      }))
  },

  render: function() {
     if (this.props.matches.length === 0){
        return (
           <div className={styles.loading}>
             <ThreeBounce />
           </div>
        );
     }

     var today = moment(24, 'HH'); // midnight tonight
     var todaysFixtures = _.filter(this.props.matches, (i) => {
        return i.kickOff.isBefore(today);
     });

     var title = 'Upcoming Fixtures';
     var fixtures = [];
     if (todaysFixtures.length > 0){
        fixtures = todaysFixtures;
        title = 'Today\'s Fixtures';
     } else {
        fixtures = _.filter(this.props.matches, (i) => {
          return i.kickOff.isAfter(today);
       });
     }

    return (
         <div>
            <h2 className={styles['fixture-list-header']}>{title}</h2>
            <div className={styles.spacer}></div>

            <ul className={styles['match-list']}>
               {fixtures.map(function(match) {
                  return <li className={match.fullTime ? styles['full-time'] : ''} key={match.id}>
                     <Link className={styles.link} to={`/match/${match.id}`}>
                        <Score match={match} />
                     </Link>
                   </li>;
                })}
            </ul>
         </div>

      );
  }
});

export default Scores;
