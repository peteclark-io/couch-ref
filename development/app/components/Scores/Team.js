'use strict'

import React from 'react';
import styles from './styles.css';

const Team = React.createClass({

   propTypes: {
      'name': React.PropTypes.string
   },

   render: function() {
      return (
         <span className={styles.teams} ref={node => (this.root = node)}>{this.props.name}</span>
      );
   }
});

export default Team;
