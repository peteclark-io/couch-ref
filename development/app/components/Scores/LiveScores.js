'use strict'

import { connect } from 'react-redux'
import Scores from './Scores'

var matches = [];

const getLiveMatches = (state = {matches: []}) => {
  return state.matches;
}

const mapStateToProps = (state) => {
  return {
    matches: getLiveMatches(state)
  }
}

const LiveScores = connect(
  mapStateToProps
)(Scores)

export default LiveScores
