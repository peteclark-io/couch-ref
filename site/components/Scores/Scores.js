'use strict'

import React from 'react';
import styles from './styles.css';

import GoalCount from './GoalCount';
import Team from './Team';

class Scores extends React.Component {

  static propTypes = {
    match: React.PropTypes.shape({
      'home': React.PropTypes.string,
      'away': React.PropTypes.string,
      'goalsHome': React.PropTypes.number,
      'goalsAway': React.PropTypes.number
    })
  };

  componentDidMount() {
    window.componentHandler.upgradeElement(this.root);
  }

  componentWillUnmount() {
    window.componentHandler.downgradeElements(this.root);
  }

  render() {
    return (
      <div ref={node => (this.root = node)}>
        <Team name={this.props.match.home} />
        <Team name={this.props.match.away} />
        <GoalCount goals={this.props.match.goalsHome} />
        <GoalCount goals={this.props.match.goalsAway} />
      </div>
    );
  }
}

export default Scores;
