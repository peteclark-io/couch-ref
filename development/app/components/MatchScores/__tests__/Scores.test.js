'use strict';

import React from 'react';
import Scores from '../Scores';
import moment from 'moment';
import renderer from 'react-test-renderer';

it('Renders two upcoming fixtures.', () => {
   const match =  {id: 'id',  home: 'home',  away: 'away',  goalsHome: 0, goalsAway: 4, kickOff: moment('2020-01-04T20:00:00Z'), televised: true, live: true}
   const match2 = {id: 'id2', home: 'home2', away: 'away2', goalsHome: 1, goalsAway: 3, kickOff: moment('2021-01-04T20:00:00Z'), televised: true, live: true}

   const rendered = renderer.create(
      <Scores matches={[match, match2]} />
   );
   expect(rendered.toJSON()).toMatchSnapshot();
});

it('Renders only televised fixtures.', () => {
   const match =  {id: 'id',  home: 'home',  away: 'away',  goalsHome: 0, goalsAway: 4, kickOff: moment('2020-01-04T20:00:00Z'), televised: true, live: true}
   const match2 = {id: 'id2', home: 'home2', away: 'away2', goalsHome: 1, goalsAway: 3, kickOff: moment('2021-01-04T20:00:00Z'), televised: false, live: true}

   const rendered = renderer.create(
      <Scores matches={[match, match2]} />
   );
   expect(rendered.toJSON()).toMatchSnapshot();
});

it('Disables links if the match is not live.', () => {
   const match =  {id: 'id',  home: 'home',  away: 'away',  goalsHome: 0, goalsAway: 4, kickOff: moment('2020-01-04T20:00:00Z'), televised: true, live: false}
   const match2 = {id: 'id2', home: 'home2', away: 'away2', goalsHome: 1, goalsAway: 3, kickOff: moment('2021-01-04T20:00:00Z'), televised: true, live: true}

   const rendered = renderer.create(
      <Scores matches={[match, match2]} />
   );
   expect(rendered.toJSON()).toMatchSnapshot();
});

it('Renders one upcoming fixture, removes other.', () => {
   const match =  {id: 'id',  home: 'home',  away: 'away',  goalsHome: 0, goalsAway: 4, kickOff: moment('2016-01-04T20:00:00Z'), televised: true, live: true}
   const match2 = {id: 'id2', home: 'home2', away: 'away2', goalsHome: 1, goalsAway: 3, kickOff: moment('2021-01-04T20:00:00Z'), televised: true, live: true}

   const rendered = renderer.create(
      <Scores matches={[match, match2]} />
   );
   expect(rendered.toJSON()).toMatchSnapshot();
});

it('Renders no fixtures if they\'re old.', () => {
   const match =  {id: 'id',  home: 'home',  away: 'away',  goalsHome: 0, goalsAway: 4, kickOff: moment('2014-01-04T20:00:00Z'), televised: true, live: true}
   const match2 = {id: 'id2', home: 'home2', away: 'away2', goalsHome: 1, goalsAway: 3, kickOff: moment('2014-02-04T20:00:00Z'), televised: true, live: true}

   const rendered = renderer.create(
      <Scores matches={[match, match2]} />
   );
   expect(rendered.toJSON()).toMatchSnapshot();
});
