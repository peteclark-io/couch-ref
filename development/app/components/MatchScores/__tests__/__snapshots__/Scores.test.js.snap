exports[`test Disables links if the match is not live. 1`] = `
<div>
  <h2
    className="fixture-list-header">
    Upcoming Fixtures
  </h2>
  <div
    className="spacer" />
  <ul
    className="match-list">
    <li
      className="">
      <div>
        <span
          className="kick-off">
          Sat 4th Jan 20:00
        </span>
        <div
          className="score">
          <span
            className="teams">
            home
          </span>
          <span
            className="goals">
            0
          </span>
        </div>
        <div
          className="score">
          <span
            className="teams">
            away
          </span>
          <span
            className="goals">
            4
          </span>
        </div>
      </div>
    </li>
    <li
      className="">
      <a
        className="link"
        onClick={[Function]}
        style={Object {}}>
        <div>
          <span
            className="kick-off">
            Mon 4th Jan 20:00
          </span>
          <div
            className="score">
            <span
              className="teams">
              home2
            </span>
            <span
              className="goals">
              1
            </span>
          </div>
          <div
            className="score">
            <span
              className="teams">
              away2
            </span>
            <span
              className="goals">
              3
            </span>
          </div>
        </div>
      </a>
    </li>
  </ul>
</div>
`;

exports[`test Renders no fixtures if they're old. 1`] = `
<div>
  <h2
    className="fixture-list-header">
    Upcoming Fixtures
  </h2>
  <div
    className="spacer" />
  <h4
    className="no-fixtures">
    No Fixtures right now! Check back later for new fixtures.
  </h4>
</div>
`;

exports[`test Renders one upcoming fixture, removes other. 1`] = `
<div>
  <h2
    className="fixture-list-header">
    Upcoming Fixtures
  </h2>
  <div
    className="spacer" />
  <ul
    className="match-list">
    <li
      className="">
      <a
        className="link"
        onClick={[Function]}
        style={Object {}}>
        <div>
          <span
            className="kick-off">
            Mon 4th Jan 20:00
          </span>
          <div
            className="score">
            <span
              className="teams">
              home2
            </span>
            <span
              className="goals">
              1
            </span>
          </div>
          <div
            className="score">
            <span
              className="teams">
              away2
            </span>
            <span
              className="goals">
              3
            </span>
          </div>
        </div>
      </a>
    </li>
  </ul>
</div>
`;

exports[`test Renders only televised fixtures. 1`] = `
<div>
  <h2
    className="fixture-list-header">
    Upcoming Fixtures
  </h2>
  <div
    className="spacer" />
  <ul
    className="match-list">
    <li
      className="">
      <a
        className="link"
        onClick={[Function]}
        style={Object {}}>
        <div>
          <span
            className="kick-off">
            Sat 4th Jan 20:00
          </span>
          <div
            className="score">
            <span
              className="teams">
              home
            </span>
            <span
              className="goals">
              0
            </span>
          </div>
          <div
            className="score">
            <span
              className="teams">
              away
            </span>
            <span
              className="goals">
              4
            </span>
          </div>
        </div>
      </a>
    </li>
  </ul>
</div>
`;

exports[`test Renders two upcoming fixtures. 1`] = `
<div>
  <h2
    className="fixture-list-header">
    Upcoming Fixtures
  </h2>
  <div
    className="spacer" />
  <ul
    className="match-list">
    <li
      className="">
      <a
        className="link"
        onClick={[Function]}
        style={Object {}}>
        <div>
          <span
            className="kick-off">
            Sat 4th Jan 20:00
          </span>
          <div
            className="score">
            <span
              className="teams">
              home
            </span>
            <span
              className="goals">
              0
            </span>
          </div>
          <div
            className="score">
            <span
              className="teams">
              away
            </span>
            <span
              className="goals">
              4
            </span>
          </div>
        </div>
      </a>
    </li>
    <li
      className="">
      <a
        className="link"
        onClick={[Function]}
        style={Object {}}>
        <div>
          <span
            className="kick-off">
            Mon 4th Jan 20:00
          </span>
          <div
            className="score">
            <span
              className="teams">
              home2
            </span>
            <span
              className="goals">
              1
            </span>
          </div>
          <div
            className="score">
            <span
              className="teams">
              away2
            </span>
            <span
              className="goals">
              3
            </span>
          </div>
        </div>
      </a>
    </li>
  </ul>
</div>
`;
